src/request.ts

```diff
@@ -28,12 +31,55 @@ export function setRequestLogger(logger: Logger): void {
 	requestLogger = logger;
 }
 
-async function _request({
-	endpoint,
-	requestBody,
-	headers: requestHeaders,
-	...options
-}: RequestOptions): Promise<unknown> {
+/**
+ * Internal function that handles retry logic for NUT-19 cached endpoints.
+ * Non-cached endpoints are executed directly without retries.
+ */
+async function requestWithRetry(options: RequestOptions): Promise<unknown> {
+	const { ttl, cached_endpoints, endpoint } = options;
+
+	const url = new URL(endpoint);
+
+	const isCachable = cached_endpoints?.some(
+		(cached_endpoint) =>
+			cached_endpoint.path === url.pathname && cached_endpoint.method === (options.method || 'GET')
+	);
```

Comment:
Should we make the default handling more explicit with `??` - ie just `null` or `undefined`?
Would surface bugs sooner if a caller accidentally passes an invalid but falsey method (eg ''. false).

```ts
... && cached_endpoint.method === (options.method ?? 'GET')
```